subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c

```diff
@@ -4,6 +4,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <soc.h>
```

Comment:
Move this after the standard includes, and before the `#include <bluetooth/hci.h>` or before the `#include "hal/cpu.h"`

Reply:
On reconsideration I moved the include file into the file that requires those macros be defined.

Reply:
We do not include header files inside internal header files in the subsystem

Reply:
> We do not include header files inside internal header files in the subsystem

Is there a justification for that?

This header is the only thing that needs access to the CMSIS macros, and it only needs it on one platform.   If you want I'll put it back, but that just makes it more likely people will remove it again in the future because the dependency is too many steps away.

Reply:
(I do not have a strong justification), other than code is cleaner across multiple header files, avoids repeated includes (and guards be handled during compilation) for internal units and as you said:
> doesn't have the necessaries elsewhere in the include paths

There are exception where headers need conditional inclusions (build and/or compile time conditional include paths/headers).

Some examples that I follow:
https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/host/hci_core.h
https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/host/hci_core.c

Some that I dislike:
https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/mesh/prov.h
https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/mesh/prov.c

Reply:
https://softwareengineering.stackexchange.com/a/167751 summaries some best practices including:

>  A TL;DR definition: A header file must include the header files that directly define each of the types directly used in or that directly declare each of the functions used in the header file in question, but must not include anything else.

By that rule it's where it belongs in the current  proposed fix.  `<soc.h>` had been included in `lll_adv.c` but was removed because it didn't seem necessary; had it been in `nordic/hal/nrf5/cpu.h` where it's actually needed the bug would never have been introduced.

Reply:
If you go by this rule and ignore the below:
> Gratuitous #include directives. Those gratuitous includes cause recompilation of things that don't need to be recompiled, and can at times make it so a system can't compile. Don't #include a file in a header if the header itself doesn't need that other header file.

I know `cpu.h` needs `soc.h`, but your inclusion of `soc.h` is gratuitous include to satisfy for other header files in `lll_adv.c` file. In other words, the following files (some to mention) too need the `#include <soc.h>` to ensure that removal of `#include "hal/.cpu.h"` will not break the compilation of `lll_adv.c`:
https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/debug.h 
https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/swi.h

I do not think we aim to have all header files in zephyr to be independently compilable. This is ok for public API header files.

Secondly, I am not able to reproduce the build error, and this PR should be closed. I think its been addressed in https://github.com/zephyrproject-rtos/zephyr/commit/9d4d6d06a0a44615b6f6f7ca9d316e24f509356a